sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/AlreadyExistException.scala

```diff
@@ -137,6 +137,15 @@ class ViewAlreadyExistsException(errorClass: String, messageParameters: Map[Stri
       messageParameters = Map("relationName" -> ident.quoted))
 }
 
+class VariableAlreadyExistsException(errorClass: String, messageParameters: Map[String, String])
+  extends AnalysisException(errorClass, messageParameters) {
```

Comment:
since we have error class, shall we just throw `AnalysisException`?

Reply:
I rote copied: ViewAlreadyExistsException(). Do we want to abandon instantiating classes like this?

Reply:
Yea, people should use error class name to differentiate errors.

Reply:
Fixed